docs/data/data-grid/list-view/ListView.js

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import { DataGridPro } from '@mui/x-data-grid-pro';
+import Box from '@mui/material/Box';
+import { useDemoData } from '@mui/x-data-grid-generator';
+import Stack from '@mui/material/Stack';
+import Avatar from '@mui/material/Avatar';
+import Typography from '@mui/material/Typography';
+import Checkbox from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import IconButton from '@mui/material/IconButton';
+import MessageIcon from '@mui/icons-material/Message';
+
+function ListViewCell(params) {
+  return (
+    <Stack
+      direction="row"
+      sx={{
+        alignItems: 'center',
+        height: '100%',
+        gap: 2,
+      }}
+    >
+      <Avatar sx={{ width: 32, height: 32, backgroundColor: params.row.avatar }} />
+      <Stack>
+        <Typography variant="body2" fontWeight={500}>
+          {params.row.name}
+        </Typography>
+        <Typography variant="body2" color="text.secondary">
+          {params.row.position}
+        </Typography>
+      </Stack>
+    </Stack>
+  );
+}
+
+const listColDef = {
+  field: 'listColumn',
+  renderCell: ListViewCell,
+};
+
+const VISIBLE_FIELDS = ['avatar', 'name', 'position'];
+
+export default function ListView() {
+  const [isListView, setIsListView] = React.useState(true);
+
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    rowLength: 20,
+    visibleFields: VISIBLE_FIELDS,
+  });
+
+  const columns = React.useMemo(() => {
+    return [
+      ...data.columns,
+      {
+        type: 'actions',
+        field: 'actions',
+        width: 75,
+        getActions: () => [
+          <IconButton aria-label="Message">
+            <MessageIcon />
+          </IconButton>,
+        ],
+      },
+    ];
+  }, [data.columns]);
+
+  const rowHeight = isListView ? 64 : 52;
+
+  return (
+    <Box sx={{ width: '100%' }}>
+      <FormControlLabel
+        control={
+          <Checkbox
+            checked={isListView}
+            onChange={(event) => setIsListView(event.target.checked)}
+          />
+        }
+        label="Enable list view"
+      />
+      <Box
+        sx={{
+          width: '100%',
+          maxWidth: isListView ? 360 : undefined,
+          height: 600,
+          margin: '0 auto',
+        }}
+      >
+        <DataGridPro
+          {...data}
+          columns={columns}
+          rowHeight={rowHeight}
+          unstable_listView={isListView}
```

Comment:
> I was wondering about listColumn, should we rename it to listViewColumn?
I felt the two names conflicting, so maybe we can help build the link between those two props.

If both of them (`unstable_listView` and `unstable_listColumn`) are required together, could we simplify further by removing unstable_listView boolean and only use listColumn / listViewColumn as a source of truth to have this feature active or not? 🤔

Reply:
In general, the less API surface, the better. But if you feel that toggling between "list view" and "normal view" is a frequent use case and the recomputation of list view state has a significant cost that should be avoided then we could surely pay a bit extra DX + bundle size cost. I'm OK with whatever you feel better. 👍

Reply:
What do others think @mui/xgrid?

Reply:
I like less API surface but I also like boolean flags for things that are boolean flags, I could be convinced either way but I think I have a small preference for keeping `listView`.

Besides, I thought this feature would automatically trigger based on media queries (under a certain width), if that's the case then the flag could act as a controlled model to override the media queries.

Reply:
> I thought this feature would automatically trigger based on media queries

Sounds like a good enhancement. If we add this, we should also provide a way for users to configure the breakpoint that gets used. Added as a point here https://github.com/mui/mui-x/issues/14997